src/en/ohjoysextoy/src/eu/kanade/tachiyomi/extension/en/ohjoysextoy/OhJoySexToy.kt

```diff
@@ -0,0 +1,135 @@
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.tryParse
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+private val MULTI_SPACE_REGEX = "\\s{6,}".toRegex()
+
+class OhJoySexToy : ParsedHttpSource() {
+
+    override val name = "Oh Joy Sex Toy"
+    override val baseUrl = "https://www.ohjoysextoy.com"
+    override val lang = "en"
+    override val supportsLatest = true
+
+    // Browse
+
+    override fun popularMangaRequest(page: Int): Request = GET("$baseUrl/category/comic/page/$page/", headers)
+
+    override fun popularMangaSelector(): String = ".comicthumbwrap"
+
+    override fun popularMangaFromElement(element: Element): SManga = SManga.create().apply {
+        setUrlWithoutDomain(element.selectFirst(".comicarchiveframe > a")!!.absUrl("href"))
+        title = element.selectFirst(".comicthumbdate")!!.text().substringBefore(" by")
+        thumbnail_url = element.selectFirst(".comicarchiveframe > a > img")?.absUrl("src")
+    }
+
+    override fun popularMangaNextPageSelector(): String = ".pagenav-left a"
+
+    // Latest
+
+    override fun latestUpdatesRequest(page: Int): Request = GET(baseUrl, headers)
+
+    override fun latestUpdatesSelector(): String = "#MattsRecentComicsBar > ul > div"
+
+    override fun latestUpdatesFromElement(element: Element): SManga = SManga.create().apply {
+        setUrlWithoutDomain(element.selectFirst(".comicarchiveframe > a")!!.absUrl("href"))
+        title = element.selectFirst(".comicthumbdate")!!.text().substringBefore(" by")
+        thumbnail_url = element.selectFirst(".comicarchiveframe > a > img")?.absUrl("src")
+    }
+
+    override fun latestUpdatesNextPageSelector(): String? = null
+
+    // Search
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request = GET("$baseUrl/?s=$query", headers)
+
+    override fun searchMangaSelector(): String = "h2.post-title"
+
+    override fun searchMangaFromElement(element: Element): SManga = SManga.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+        title = element.selectFirst("a")!!.text().substringBefore(" by")
+    }
+
+    override fun searchMangaNextPageSelector(): String? = null
+
+    // etc
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        thumbnail_url = document.selectFirst("meta[property=\"og:image\"]")
+            ?.absUrl("content")
+        status = SManga.COMPLETED
+        title = document.selectFirst("meta[property=\"og:title\"]")
+            ?.attr("content")
+            ?.substringBefore(" by")
+            ?: ""
```

Comment:
should not fall back to empty
```suggestion
        title = document.selectFirst("meta[property=\"og:title\"]")!!
            .attr("content")
            .substringBefore(" by")
```